src/rpc/net.cpp

```diff
@@ -228,17 +228,16 @@ static UniValue addnode(const JSONRPCRequest& request)
     std::string strCommand;
     if (!request.params[1].isNull())
         strCommand = request.params[1].get_str();
-    if (request.fHelp || request.params.size() != 2 ||
+    if (request.fHelp || request.params.size() < 2 || request.params.size() > 3 ||
```

Comment:
#18606 tries to unify the coding-style of all non-hidden RPC calls by using `RPCHelpMan::Check`

If #18606 gets merged before this PR, this change shouldn't be necessary (should be automatically handled by `RPCHelpMan::IsValidNumArgs`). Only the line adding the parameter `privileged` below will be necessary.

And thanks to #18606 the new bash auto-completion file will be generated automatically by  running  `test/functional/tool_cli_completion.py --overwrite` before commiting :smiley: 

If this PR gets merged before #18606 I'll update my PR accordingly.